habitat/datasets/rearrange/samplers.py

```diff
@@ -525,6 +567,27 @@ def sample(
             # now find a matching link
             for link_ix in range(ao_instance.num_links):
                 if ao_instance.get_link_name(link_ix) == self.link_name:
+                    if receptacle is not None:
+                        if (
+                            ao_instance.handle
+                            == receptacle.parent_object_handle
+                        ) and (
+                            (link_ix == receptacle.parent_link)
+                            or ("frige" in receptacle.parent_object_handle)
+                            or ("fridge" in receptacle.parent_object_handle)
+                        ):
+                            # If this is true, this means that the receptacle AO must be opened. That is because
+                            # the object is spawned inside the fridge OR inside the kitchen counter BUT not on top of the counter
+                            # because in this case all drawers must be closed.
+                            joint_state = random.uniform(
+                                self.state_range[0], self.state_range[1]
+                            )
+                        else:
+                            joint_state = 0.0
```

Comment:
This assumes 0 joint state is default. This isn't necessarily the case for generic URDF. Since we allow state overrides in the scene config, it would be best to use the current state of joints upon initialization.